Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=7aed55d1085f71241284a30af0300feea48c36db
Commit:     7aed55d1085f71241284a30af0300feea48c36db
Parent:     37dd3cb4153084ff693e738b7ee087990d351e5e
Author:     Jiri Slaby <jirislaby@gmail.com>
AuthorDate: Sun Jan 4 00:27:09 2009 +0100
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Sun Jan 4 10:20:29 2009 +0100

    x86: fix RIP printout in early_idt_handler
    
    Impact: fix debug/crash printout
    
    Since errorcode is popped out, RIP is on the top of the stack.
    Use real RIP value instead of wrong CS.
    
    Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
    Cc: <stable@kernel.org>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

 arch/x86/kernel/head_64.S |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/x86/kernel/head_64.S b/arch/x86/kernel/head_64.S
index 26cfdc1..0e275d4 100644
--- a/arch/x86/kernel/head_64.S
+++ b/arch/x86/kernel/head_64.S
@@ -305,7 +305,7 @@ ENTRY(early_idt_handler)
 	call dump_stack
 #ifdef CONFIG_KALLSYMS	
 	leaq early_idt_ripmsg(%rip),%rdi
-	movq 8(%rsp),%rsi	# get rip again
+	movq 0(%rsp),%rsi	# get rip again
 	call __print_symbol
 #endif
 #endif /* EARLY_PRINTK */
